cs/src/core/Index/FasterLog/FasterLog.cs

```diff
@@ -179,6 +180,40 @@ public unsafe bool TryAppend(Span<byte> entry, ref long logicalAddress)
             return true;
         }
 
+        private TaskCompletionSource<long> commitTask = new TaskCompletionSource<long>();
```

Comment:
Please use TaskCreationOptions.RunContinuationsAsynchronously to avoid potential deadlocks in consuming code?